terraform/node_resource_plan.go

```diff
@@ -177,20 +177,51 @@ func (n *NodePlannableResource) ModuleInstance() addrs.ModuleInstance {
 	return n.Addr.Module
 }
 
-// GraphNodeEvalable
-func (n *NodePlannableResource) EvalTree() EvalNode {
+// GraphNodeExecutable
+func (n *NodePlannableResource) Execute(ctx EvalContext, op walkOperation) error {
 	if n.Config == nil {
 		// Nothing to do, then.
 		log.Printf("[TRACE] NodeApplyableResource: no configuration present for %s", n.Name())
-		return &EvalNoop{}
+		return nil
 	}
 
-	// this ensures we can reference the resource even if the count is 0
-	return &EvalWriteResourceState{
-		Addr:         n.Addr,
-		Config:       n.Config,
-		ProviderAddr: n.ResolvedProvider,
+	var diags tfdiags.Diagnostics
```

Comment:
This duplicate chunk of code could probably be delegated to a single function, or a method on the `NodeAbstractResource` type. This one isn't too large, but I'd like to avoid copying the body of the eval nodes into each of their callers, then forgetting about some copies.